### PR TITLE
Changed markup when using optgroup element

### PR DESCRIPTION
**Pull request by lucascumsille**
Instead of using the pseudo element to display the title of a optgroup, now they will be wrapped on a fieldset with a legend. The legend will have the text from the label atrribute from the optgroup.

It's also possible to change the class names for the fieldset and legend, by using these settings: menuFieldsetHTML and menuFieldsetLegendHTML.

Screenshot 2024-01-15 at 14 10 36

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -74,7 +74,8 @@
         $('#people').multiSelect();
         $('#ice-cream').multiSelect();
         $('#line-wrap-example').multiSelect({
-            positionMenuWithin: $('.position-menu-within')
+            positionMenuWithin: $('.position-menu-within'),
+            menuItemHTML: $('<label class="multi-select-menuitem test">'),
```
Comment (dracos) [inline demo/index.html:78]: Presumably this change not needed :)

File: src/jquery.multi-select.min.js
Comment (dracos) [inline src/jquery.multi-select.min.js:1]: This file has disappeared

File: src/example-styles.css [hunk 2]
```diff
@@ -24,25 +24,11 @@
...
-.multi-select-menuitem--titled:before {
-    display: block;
-    font-weight: bold;
-    content: attr(data-group-title);
-    margin: 0 0 0.25em -20px;
-}
-
-.multi-select-menuitem--titledsr:before {
-    display: block;
+legend {
```
Comment (dracos) [inline src/example-styles.css:27]: The rest of the file only uses class names, probably good to stick with that, and use `.multi-select-legend` here?

File: src/example-styles.css [hunk 3]
```diff
@@ -24,25 +24,11 @@
...
-    border: 0;
-    clip: rect(0 0 0 0);
-    height: 1px;
-    margin: -1px;
-    overflow: hidden;
-    padding: 0;
-    position: absolute;
-    width: 1px;
+    padding-left: 10px;
+    padding-top: 0.25rem;
```
Comment (dracos) [inline src/example-styles.css:31]: Dunno if this is just me used to the old one, but this felt a bit spaced out to me now, especially if there's going to be a lot of options? What do you think about dropping this padding-top and adding a:
`.multi-select-legend + .multi-select-menuitem { padding-top: 0.25rem; }`
to reduce the spacing a bit?

File: src/jquery.multi-select.js [hunk 4]
```diff
@@ -292,16 +296,18 @@
 
     constructMenuItemsGroup: function($optgroup, optgroup_index) {
       var _this = this;
+      var $fieldset = $(_this.settings['menuFieldsetHTML']);
+      console.log($fieldset);
```
Comment (dracos) [inline src/jquery.multi-select.js:300]: will need removing

**Review by dracos — CHANGES_REQUESTED** [review]
Some suggestions, let me know what you think. Also as mentioned elsewhere we'll need to make the up/down changes here (in e.g. the upDown function).

**Comment by dracos** [comment]
@lucascumsille I've pushed a commit that hopefully gets up/down working as before if you'd like to take a look. I've also fixed the tests.

**Comment by lucascumsille** [comment]
@dracos That's perfect is working for me. I tried it on Voiceover, and it works slightly different than the one we used to have, where on each option it would read = "Group title + option name" Example "Flavour Vanilla", now it gives the name of the group when you are in the first option of the group, which is exactly what happens with GOVUK and by default.

If we wanted to keep the current behaviour we could include an `aria-label` that reads "Flavour Vanilla", but I'm not sure if it's unnecessary, because screenreader users are already used to the default. At the same time I liked the fact that "Flavour Vanilla" provides context without the need of having good memory, but it can be a pain with long lists or fieldsets with long names.

Let me know what you think, otherwise I can just merge to master.

**Comment by dracos** [comment]
I think going with the default fieldset behaviour is probably more consistent, yeah - the category list on FMS map filter can be quite long, but OTOH will have a few fieldsets in it, but hopefully will be clear.

**Comment by lucascumsille** [comment]
@dracos Sounds good, let's keep it like that. Should I merge it master?

**Comment by dracos** [comment]
yeah, go for it, up to you if you want to merge the two commits as one or keep it as it is

**Pull request merged**